src/main/NWNX/API/Events/PartyEvents.cs

```diff
@@ -0,0 +1,329 @@
+
+      protected override void PrepareEvent(NwObject objSelf)
+      {
+        Player = (NwPlayer)objSelf;
+        IgnoredBy = NWScript.StringToObject(EventsPlugin.GetEventData("INVITED_BY")).ToNwObject<NwCreature>();
+      }
+    }
+
+    [NWNXEvent("NWNX_ON_ACCEPT_INVITATION_BEFORE")]
+    public class OnAcceptInvitationBefore : EventSkippable<OnAcceptInvitationBefore>
+    {
+      /// <summary>
+      /// Gets the player doing the action.
+      /// </summary>
+      public NwPlayer Player { get; private set; }
+
+      /// <summary>
+      /// Gets NwCreature that accepted invitation.
+      /// </summary>
+      public NwCreature AcceptedBy { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf)
+      {
+        Player = (NwPlayer)objSelf;
+        AcceptedBy = NWScript.StringToObject(EventsPlugin.GetEventData("INVITED_BY")).ToNwObject<NwCreature>();
+      }
+    }
+
+    [NWNXEvent("NWNX_ON_ACCEPT_INVITATION_AFTER")]
+    public class OnAcceptInvitationAfter : EventSkippable<OnAcceptInvitationAfter>
+    {
+      /// <summary>
+      /// Gets the player doing the action.
+      /// </summary>
+      public NwPlayer Player { get; private set; }
+
+      /// <summary>
+      /// Gets NwCreature that accepted invitation.
+      /// </summary>
+      public NwCreature AcceptedBy { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf)
+      {
+        Player = (NwPlayer)objSelf;
+        AcceptedBy = NWScript.StringToObject(EventsPlugin.GetEventData("INVITED_BY")).ToNwObject<NwCreature>();
+      }
+    }
+
+    [NWNXEvent("NWNX_ON_REJECT_INVITATION_BEFORE")]
+    public class OnRejectInvitationBefore : EventSkippable<OnRejectInvitationBefore>
+    {
+      /// <summary>
+      /// Gets the player doing the action.
+      /// </summary>
+      public NwPlayer Player { get; private set; }
+
+      /// <summary>
+      /// Gets NwCreature that rejected the invitation.
+      /// </summary>
+      public NwCreature RejectedBy { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf)
+      {
+        Player = (NwPlayer)objSelf;
+        RejectedBy = NWScript.StringToObject(EventsPlugin.GetEventData("INVITED_BY")).ToNwObject<NwCreature>();
+      }
+    }
+
+    [NWNXEvent("NWNX_ON_REJECT_INVITATION_AFTER")]
+    public class OnRejectInvitationAfter : EventSkippable<OnRejectInvitationAfter>
+    {
+      /// <summary>
+      /// Gets the player doing the action.
+      /// </summary>
+      public NwPlayer Player { get; private set; }
+
+      /// <summary>
+      /// Gets NwCreature that rejected the invitation.
+      /// </summary>
+      public NwCreature RejectedBy { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf)
+      {
+        Player = (NwPlayer)objSelf;
+        RejectedBy = NWScript.StringToObject(EventsPlugin.GetEventData("INVITED_BY")).ToNwObject<NwCreature>();
+      }
+    }
```

Comment:
We should test/ask around for what ObjectSelf is for these events, and update the property naming/comments to match.

I would only include the EventData property for now, and draft these other properties in a future PR.